### PR TITLE
add several resolution test cases

### PR DESCRIPTION
**Pull request by bvanjoi**
r? @petrochenkov

### CODE REVIEW

File: tests/ui/imports/glob-conflict-cross-crate-3.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+mod a {
+    pub type C = i32;
+}
+
+use glob_conflict_cross_crate_2_extern::*;
+use a::*;
+
+fn main() {
+    let _a: C = 1;
```
Comment (petrochenkov) [inline tests/ui/imports/glob-conflict-cross-crate-3.rs:14]:
```suggestion
    let _a: C = 1; // FIXME this use should be ambiguous
```
Could you add FIXME comments to all lines that get ERROR/WARN annotations in https://github.com/rust-lang/rust/pull/114682?
Including old tests like `tests/ui/imports/extern-with-ambiguous-3.rs`.

**Comment by petrochenkov** [comment]
The `tests/ui/imports/ambiguous-4.rs` change can also be moved to this PR.
@rustbot author

File: tests/ui/imports/ambiguous-4.rs [hunk 2]
```diff
@@ -5,5 +5,4 @@ extern crate ambiguous_4_extern;
 
 fn main() {
     ambiguous_4_extern::id();
-    // `warning_ambiguous` had been lost at metadata.
```
Comment (bvanjoi) [inline tests/ui/imports/ambiguous-4.rs:8]: It appears that `﻿ambiguous_4_extern::id` should be considered an ambiguous item, however, this isn't currently implemented in #114682.

**Comment by bvanjoi** [comment]
Do you think we should create an issue addressing the edge case mentioned in https://github.com/rust-lang/rust/pull/114682#issuecomment-1881630225?

File: tests/ui/imports/ambiguous-4.rs (hunk 2 above)
Comment (petrochenkov) [reply tests/ui/imports/ambiguous-4.rs:8]: Is it due to this case https://github.com/rust-lang/rust/pull/114682#discussion_r1455745546 or something different?

**Comment by petrochenkov** [comment]
Maybe, I just don't understand what is the issue with https://github.com/rust-lang/rust/pull/114682#issuecomment-1881630225.
The import should refer to *both* `extern_crate::d::max` and `extern_crate::e::max` in different namespaces, but that comment only considers one namespace.

File: tests/ui/imports/ambiguous-4.rs (hunk 2 above)
Comment (petrochenkov) [reply tests/ui/imports/ambiguous-4.rs:8]: In any case, if this should be an ambiguity, then the same FIXME comment as in other tests would be appropriate.

**Comment by petrochenkov** [comment]
In any case, adding a test for this would be useful.

**Comment by bvanjoi** [comment]
Additional details have been appended to `﻿issue-114682-2.rs`.

**Pull request merged**